core/src/main/java/io/grpc/ClientInterceptors.java

```diff
@@ -115,70 +115,102 @@ private ProcessInterceptorChannel(Channel channel, Iterable<ClientInterceptor> i
   /**
    * A {@link Call} which forwards all of it's methods to another {@link Call}.
    */
-  public static class ForwardingCall<ReqT, RespT> extends Call<ReqT, RespT> {
+  public abstract static class ForwardingCall<ReqT, RespT> extends Call<ReqT, RespT> {
 
-    private final Call<ReqT, RespT> delegate;
-
-    public ForwardingCall(Call<ReqT, RespT> delegate) {
-      this.delegate = delegate;
-    }
+    /**
+     * Returns the delegated {@code Call}
+     */
+    protected abstract Call<ReqT, RespT> delegate();
 
     @Override
     public void start(Listener<RespT> responseListener, Metadata.Headers headers) {
-      this.delegate.start(responseListener, headers);
+      this.delegate().start(responseListener, headers);
     }
 
     @Override
     public void request(int numMessages) {
-      this.delegate.request(numMessages);
+      this.delegate().request(numMessages);
     }
 
     @Override
     public void cancel() {
-      this.delegate.cancel();
+      this.delegate().cancel();
     }
 
     @Override
     public void halfClose() {
-      this.delegate.halfClose();
+      this.delegate().halfClose();
     }
 
     @Override
     public void sendPayload(ReqT payload) {
-      this.delegate.sendPayload(payload);
+      this.delegate().sendPayload(payload);
     }
   }
 
+  private static final Call<Object, Object> NOOP_CALL = new Call<Object, Object>() {
```

Comment:
nm. I'll comment here.

Yeah, CheckedForwardingCall looks like our best solution.

By the looks of things, it would be useful to have a SimpleForwardingCall (similar to what is [done in Guava](http://docs.guava-libraries.googlecode.com/git-history/release/javadoc/com/google/common/util/concurrent/ForwardingFuture.SimpleForwardingFuture.html)) that most call sites use.

I'm thinking we probably will want to make ForwardingCall a top-level class. We could leave a @Deprecated class here that just extends SimpleForwardingFuture. That is pretty easy and doesn't break builds, but still allows people to swap to the new API.

We can discuss in person.